acra/guides/advanced-integrations/client-side-integration-with-acra-connector.md

```diff
@@ -0,0 +1,224 @@
+
+However, if your code is a huge family of microservices where some of them just write data and some just read it, it is essential to be able to do it the way you want it. You can write AcraStructs generated by AcraWriter directly into the database so that you don't need to carry all the AcraConnector keys with every piece of code on every machine.
+
+For example, we're aware of a setup where AcraStructs are sent down the Kafka stream and emerge in the database at some further point in the future.
+
+### Client-side without Zones
+
+This is an instruction for PostgreSQL.
+
+#### Writes
+
+When you need to store some sensitive data:
+
+- before doing `INSERT`/`UPDATE` on data, you should use AcraWriter libraries
+  specific to your language, encrypting sensitive data into AcraStruct with AcraServer's
+  public key with suffix `_storage` (`some_client_id_storage.pub`) that is different from the keypair used by AcraConnector for communicating with AcraServer.
+- then you can proceed with doing `INSERT`/`UPDATE`, either through AcraConnector or directly.
+
+#### Reads
+
+Read requests are the same as the regular reads, but their remote address is AcraConnector instead of your database listener. A typical PostgreSQL connection address would look like this: `postgresql://user:password@127.0.0.1:9494/db_name`.
+
+> Note for PostgreSQL: In the current layout, you aren't required to use SSL when trying to connect to the database. Transport protection for sensitive (decrypted) data is provided between AcraServer and AcraConnector via [Secure Session]({{<ref "themis/crypto-theory/cryptosystems/secure-session.md" >}}). However, you can setup using SSL connection, too.
+
+### Client-side with Zones
+
+Description of the client-side with Zones for PostgreSQL.
+
+Zone-based encryption is the best way to cryptographically compartmentalise the data that comes from different sources, following the user's choice. Cryptographically, the Zones are a mechanism for modifying the AcraServer key according to the Zone Id and for mapping Zone's private keys during the decryption. In theory, you could use a new key for every record. However, it would introduce an unwanted overhead on AcraServer's performance.
+
+#### Writing
+
+To write AcraStruct with Zone keys, you need to:
+
+* generate the Zone keys on AcraServer,
+* acquire Zone public key and Zone Id from AcraServer,
+* encrypt data using this key as AcraServer key in a standard AcraStruct process.
+
+There are two ways you can generate the keys:
+
+- First, make sure that you already generated master key and exported it as ACRA_MASTER_KEY environmental variable, or generate it like this:
+```
+export ACRA_MASTER_KEY=$(echo -n "My_Very_Long_Key_Phrase_ge_32_chars" | base64)
+```
+Read more about different types of keys used in Acra on [Key Management]({{< ref "acra/acra-in-depth/cryptography-and-key-management/#-INVALID" >}}) page.
```

Comment:
I don't think it's possible to launch AcraServer without a master key. You need to use AcraServer's master key when adding new zones. The instructions should tell this.

Reply:
I re-structured and placed it as part of `acra-addzone` usage case. Because the user doesn't need it when will use AcraConnector's HTTP APi. After that, there are correct 2 ways to generate keys.